lib/pbench/server/cache_manager.py

```diff
@@ -274,59 +277,51 @@ def cache_map(self, tar_path: Path):
             for l_path in dir_list:
                 tar_info = FileInfo(l_path)
                 curr[l_path.name] = {"details": tar_info}
-                # Note: have to handle symlink case as it
-                # is not being checked whether it is a symlink,
-                # it is checking for the redirected path. Currently,
-                # as I have used file for the symlink it is not added it
-                # the queue, but if it a symlink is pointing to a directory
-                # it turns out to be an endless loop. There may be case
-                # that the way I am creating symlink in tests is wrong because
-                # when I am testing manually I am not seeing the same case.
                 if l_path.is_dir():
                     dir_queue.append((l_path, curr))
         self.cachemap = cmap
 
     @staticmethod
-    def traverse_cmap(path, cachemap):
+    def traverse_cmap(path: Path, cachemap: dict):
+        name = path.name
         files_list = path.parts
         c_map = cachemap
 
         for file_l in files_list:
-            if file_l != ["/", "//"]:
+            if file_l not in ("/", "//"):
                 c_map = c_map[file_l]
                 if c_map["details"].type == "directory":
-                    if "children" in c_map:
+                    if "children" in c_map and file_l != name:
                         c_map = c_map["children"]
-                elif c_map["details"].type == "symlink":
-                    resolve_path = c_map["details"].resolve_path
-                    c_map = Tarball.traverse_cmap(resolve_path, cachemap)
 
         return c_map
```

Comment:
The problem with `path.parts` is if the path is starting with a "/" only then it will consiter adding `/` in it's list else it will skip it. So, if the path is "/mock/dir_name" the list will be `('/', 'mock', 'dir_name')` and if it is "mock/dir_name" then it will be `('mock', 'dir_name')` . To take care of both the cases I added if condition instead of skipping 1st value of list directly.

But yeah the solution that you provided also takes care of that

Reply:
what if the path is a file.. we don't want to go till the last second element of the list we want to go till the last and that is where we can find the deatils of the file.